code/game/objects/items/weapons/twohanded.dm

```diff
@@ -121,10 +122,14 @@
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
 
 /obj/item/weapon/twohanded/offhand/unwield()
-	qdel(src)
+	if(wielded)//Only delete if we're wielded
+		wielded = FALSE
+		qdel(src)
```

Comment:
Question on why you did this, actually.

Why check for this, instead of just doing this?

```
/obj/item/weapon/twohanded/offhand/unwield()
	if(!QDELETED(src)
		qdel(src)
```

Reply:
checking for QDELETED is a code smell in my opinion, 99% of the time your code should be structured to avoid recursive deletes

Reply:
Hmmm. You sure about that? Plenty of things use `QDELETED` for exactly this situation, only in different context. Anything that uses `do_after` typically checks for this and timers check for this, for instance.

While all of those could be worked around by using variables in a clever manner (as you have done here), or utilizing new variables, is that really inherently better than just using `QDELETED`, which was pretty much made for this exact purpose?

Reply:
apologies I mean the pattern of
```
if not qdeleted then
 qdelete
```

Reply:
That's what I meant, really; there's other things that use that in the code; it's not, at all, uncommon; particularly with the timer system.

Reply:
then they're poorly structured

Reply:
That's ridiculous. Someone having to have a unique var and structure in such a way (through the entire code) just so it doesn't potentially `Destroy` loop would be almost *impossible* for even the most skilled of coders to account for. Dare I say virtually impossible.